app/components/support_interface/application_choice_component.rb

```diff
@@ -10,92 +10,139 @@ def initialize(application_choice)
       end
+    end
 
-      if application_choice.rejected?
-        if application_choice.rejected_by_default
-          rows << { key: 'Rejected by default at', value: application_choice.rejected_at.to_s(:govuk_date_and_time) }
-        else
-          rows << { key: 'Rejected at', value: application_choice.rejected_at.to_s(:govuk_date_and_time) }
-        end
-        if rejection_reasons_text
-          rows << { key: 'Rejection reason', value: rejection_reasons_text }
-        end
+    def offer_made_at_row
+      return unless application_choice.offer?
+
+      { key: 'Offer made at', value: application_choice.offered_at.to_s(:govuk_date_and_time) }
+    end
+
+    def decline_by_default_at_row
+      return unless application_choice.offer?
+
+      { key: 'Decline by default at', value: application_choice.decline_by_default_at.to_s(:govuk_date_and_time) } if application_choice.decline_by_default_at
+    end
+
+    def course_candidate_applied_for_row
+      return unless application_choice.different_offer?
+
+      { key: 'Course candidate applied for', value: render(CourseOptionDetailsComponent.new(course_option: application_choice.course_option)) }
+    end
+
+    def course_offered_by_provider_row
+      return unless application_choice.different_offer?
+
+      { key: 'Course offered by provider', value: render(CourseOptionDetailsComponent.new(course_option: application_choice.current_course_option)) }
+    end
+
+    def course_row
+      return if application_choice.different_offer?
+
+      { key: 'Course', value: render(CourseOptionDetailsComponent.new(course_option: application_choice.course_option)) }
+    end
+
+    def rejected_at_or_by_default_at_row
+      return unless application_choice.rejected?
+
+      if application_choice.rejected_by_default
+        { key: 'Rejected by default at', value: application_choice.rejected_at.to_s(:govuk_date_and_time) }
+      else
+        { key: 'Rejected at', value: application_choice.rejected_at.to_s(:govuk_date_and_time) }
       end
+    end
 
-      rows << { key: 'Sent to provider at', value: application_choice.sent_to_provider_at.to_s(:govuk_date_and_time) } if application_choice.sent_to_provider_at
-      rows << { key: 'Reject by default at', value: application_choice.reject_by_default_at.to_s(:govuk_date_and_time) } if application_choice.reject_by_default_at && application_choice.decision_pending?
-      rows << { key: 'Decline by default at', value: application_choice.decline_by_default_at.to_s(:govuk_date_and_time) } if application_choice.decline_by_default_at && application_choice.offer?
+    def rejection_reason_row
+      return unless rejection_reasons_text
 
+      { key: 'Rejection reason', value: rejection_reasons_text }
+    end
+
+    def sent_to_provider_at_row
+      { key: 'Sent to provider at', value: application_choice.sent_to_provider_at.to_s(:govuk_date_and_time) } if application_choice.sent_to_provider_at
+    end
+
+    def reject_by_default_at_row
```

Comment:
I think this is covered on line 84 right?

Reply:
Not quite... as I understand `reject_by_default_at` is a date in the future that is generated when app choice is decision pending.

Line 84 displays `rejected_at` which is a date generic to any rejected application choice. The logic around this draws from `rejected_by_default` which is a boolean set to true when an application choice is rejected by default.

TL;DR - This is present/future tense theother is past tense and they pull dates from different db columns.

Reply:
Right, sorry about that i missed the tense 👍  Good to go then